xarray/core/duck_array_ops.py

```diff
@@ -190,11 +191,7 @@ def lazy_array_equiv(arr1, arr2):
     arr2 = asarray(arr2)
     if arr1.shape != arr2.shape:
         return False
-    if (
-        dask_array
-        and isinstance(arr1, dask_array_type)
-        and isinstance(arr2, dask_array_type)
-    ):
+    if dask_array and is_duck_dask_array(arr1) and is_duck_dask_array(arr2):
```

Comment:
Do duck dask arrays have a unique token saved in `name`? Doesn't seem so in https://github.com/hgrecco/pint/pull/1129/files

Reply:
No, they don't. Dask recommends implementing deterministic hashing with `__dask_tokenize__` when [defining custom collections](https://docs.dask.org/en/latest/custom-collections.html#deterministic-hashing), but I refrained from doing so. Pint seems to use tokens, but I'm not familiar with how they work or how they would interact with Dask. Could this cause problems?

Reply:
Yes the `arr1.name == arr2.name` check depends on the `.name` property having a unique token. Then we can test equality cheaply. This is necessary because xarray likes to compare "non-dimensional coordinate variables" for equality in many places. This can lead to major slowdowns if we can't do this check cheaply.

One options is:
1. update this check to use `tokenize` instead of the `name` property for duck dask arrays.
2. for actual dask arrays we could use `.name` as an optimization, instead of calculating the token

Reply:
Also it would be a good idea to define `__dask_tokenize__` in pint. It helps prevent unnecessary duplicate computations.

> Pint seems to use tokens,

What are these tokens?

Reply:
> Also it would be a good idea to define `__dask_tokenize__` in pint. It helps prevent unnecessary duplicate computations.

Okay, I can open an issue and work on that.

> What are these tokens?

They're from the `token` library and if I understand correctly they're used to resolve unit operations. I think this is different than how Dask uses tokens and `.name`. If someone knows more about this please feel free to correct me.